### PR TITLE
create lesson

### PR DESCRIPTION
**Pull request by AhsanBracu**

### CODE REVIEW

File: app/src/main/java/com/framgia/elsytem/UpdateProfileActivity.java [hunk 1]
```diff
@@ -173,4 +201,125 @@ protected void onPostExecute(String result) {
...
+        jsonObject.accumulate("category_id",1);
+        jsonObject.accumulate("option","all_word");
+        jsonObject.accumulate("page", "1");
+        // parent.put(jsonObject);
+        result= jsonObject.toString();
+        return  result;
+    }
+    public String createLesson() throws JSONException {
+        String result;
+        JSONObject parent = new JSONObject();
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/UpdateProfileActivity.java]: using Gson please

Comment (AhsanBracu) [inline app/src/main/java/com/framgia/elsytem/UpdateProfileActivity.java]: Ok

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java [hunk 2]
```diff
@@ -30,22 +60,97 @@ protected void onCreate(Bundle savedInstanceState) {
...
         getSupportActionBar().setHomeButtonEnabled(true);
         getSupportActionBar().setTitle("Basic Lesson");
+        option_1 = (Button) findViewById(R.id.choice_1);
+        option_2 = (Button) findViewById(R.id.choice_2);
+        option_3 = (Button) findViewById(R.id.choice_3);
+        option_4 = (Button) findViewById(R.id.choice_4);
+        txttword = (TextView) findViewById(R.id.word);
+        okHttpClient = new OkHttpClient();
         mTextToSpeech = new TextToSpeech(this, this);
+        Toast.makeText(getApplication(), "value i" +i, Toast.LENGTH_LONG).show();
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: why toast here? this is training project not sample project.

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java [hunk 3]
```diff
@@ -30,22 +60,97 @@ protected void onCreate(Bundle savedInstanceState) {
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
         getSupportActionBar().setHomeButtonEnabled(true);
         getSupportActionBar().setTitle("Basic Lesson");
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: no hard code please

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java [hunk 4]
```diff
@@ -30,22 +60,97 @@ protected void onCreate(Bundle savedInstanceState) {
...
             public void onClick(View v) {
                 speak();
-                startActivity(new Intent(getApplication(),ResultActivity.class));
+                startActivity(new Intent(getApplication(), ResultActivity.class));
+            }
+        });
+        option_1.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Toast.makeText(getApplication(), "click", Toast.LENGTH_LONG).show();
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: why toast?

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java [hunk 5]
```diff
@@ -30,22 +60,97 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                z=3;
+                createalert();
+            }
+        });
+        PREF_STATE = getApplicationContext().getSharedPreferences("MyPref",
+                MODE_PRIVATE);
+        int pref_State = PREF_STATE.getInt("state", 0);
+        new HttpAsyncLesson().execute(" https://manh-nt.herokuapp.com/categories/1/lessons.json");
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: create class to keep constant value like this please

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java [hunk 6]
```diff
@@ -30,22 +60,97 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                            option_2.setText(answersEntityArrayList.get(1).getContent());
+                            option_3.setText(answersEntityArrayList.get(2).getContent());
+                            option_4.setText(answersEntityArrayList.get(3).getContent());
+                        }
+                        editor = PREF_STATE.edit();
+                        editor.putInt("state",i);
+                        editor.commit();
+                    }
+                });
+        aBuilder.setNegativeButton("No", new DialogInterface.OnClickListener() {
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: do not hard code please

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java [hunk 7]
```diff
@@ -30,22 +60,97 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            }
+        });
+        PREF_STATE = getApplicationContext().getSharedPreferences("MyPref",
+                MODE_PRIVATE);
+        int pref_State = PREF_STATE.getInt("state", 0);
+        new HttpAsyncLesson().execute(" https://manh-nt.herokuapp.com/categories/1/lessons.json");
+    }
+    public void createalert() {
+        final AlertDialog.Builder aBuilder = new AlertDialog.Builder(QuestionActivity.this);
+        aBuilder.setMessage("Do you want to continue with this answer???");
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: hardcode

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java [hunk 8]
```diff
@@ -1,26 +1,53 @@
...
     ImageButton play;
+    OkHttpClient okHttpClient;
+    Lesson lesson;
+    public static SharedPreferences.Editor editor = null;
+    public static final MediaType JSON
+            = MediaType.parse("application/json; charset=utf-8");
+    Button option_1, option_2, option_3, option_4;
+    TextView txttword;
+    Gson gson;
+    int x = 1;
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: please naming this parameter clearly, i dont know what is x, i, z

**Pull request merged**